tests/hw-accel/kmm/modules/tests/signing-test.go

```diff
@@ -217,19 +218,21 @@ var _ = Describe("KMM", Ordered, Label(kmmparams.LabelSuite, kmmparams.LabelSani
 				Create()
 			Expect(err).ToNot(HaveOccurred(), "error while creating preflight")
 
-			By("Await build pod to complete build")
-			err = await.BuildPodCompleted(APIClient, kmmparams.ModuleBuildAndSignNamespace, 5*time.Minute)
-			Expect(err).ToNot(HaveOccurred(), "error while building module")
-
 			By("Await preflightvalidationocp checks")
+			// account for pull and build on different clusters
 			err = await.PreflightStageDone(APIClient, kmmparams.PreflightName, moduleName,
-				kmmparams.ModuleBuildAndSignNamespace, time.Minute)
-			Expect(err).To(HaveOccurred(), "preflightvalidationocp did not complete")
+				kmmparams.ModuleBuildAndSignNamespace, 3*time.Minute)
+			Expect(err).NotTo(HaveOccurred(), "preflightvalidationocp did not complete")
+
+			By("Await build pod to complete build")
```

Comment:
Await on build pod should be before the await.PreflightStageDone. 

Once the Preflight is created, it should start a build which the `await.BuildPodCompleted` should monitor. Then it gets the preflightvalidation status. with `await.PreflightStageDone`

Still with KMM 2.4.1 there is a bug related to preflight that even if the imagePush is set to false, it ignores it and it pushes the image to the registry.